src/librustc_interface/passes.rs

```diff
@@ -352,13 +352,7 @@ fn configure_and_expand_inner<'a>(
         )
     });
 
-    // If we're actually rustdoc then there's no need to actually compile
-    // anything, so switch everything to just looping
-    let mut should_loop = sess.opts.actually_rustdoc;
     if let Some(PpMode::PpmSource(PpSourceMode::PpmEveryBodyLoops)) = sess.opts.pretty {
-        should_loop |= true;
-    }
-    if should_loop {
         log::debug!("replacing bodies with loop {{}}");
         util::ReplaceBodyWithLoop::new(&mut resolver).visit_crate(&mut krate);
     }
```

Comment:
Why not completely remove this?

Reply:
This is still useful as a debugging option, originally it had nothing to do with rustdoc: https://github.com/rust-lang/rust/pull/19964.

I don't feel strongly about it but I don't think removing it should go in the same PR as fixing rustdoc bugs.

Reply:
I'm not too convinced by https://github.com/rust-lang/rust/pull/19964.
I'd suggest to open a PR removing it and poll people on whether anybody actually used it for debugging it since 2014.

Reply:
cc @pnkfelix, I think they said they still use it.

Reply:
I wrote a blog post in late 2019 that demonstrated use of `everybody_loops`. So yes, I do still use it.

http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/#L..loopification.....via.pretty-printer